bcache/bcache.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bcache provides access to statistics exposed by the bcache (Linux
+// block cache).
+package bcache
+
+// Stats contains bcache runtime statistics, parsed from /sys/fs/bcache/.
+//
+// The names and meanings of each statistic were taken from bcache.txt and
+// files in drivers/md/bcache in the Linux kernel source. Counters are float64
+// (in-kernel counters are mostly unsigned long).
+type Stats struct {
+	// The name of the bcache used to source these statistics.
+	Name   string
+	Bcache BcacheStats
+	Bdevs  []BdevStats
+	Caches []CacheStats
+}
+
+// BcacheStats contains statistics tied to a bcache ID.
+type BcacheStats struct {
+	AverageKeySize        float64
```

Comment:
Will the source ever return floats, or only integers?

While Prometheus only supports float64 as data type, the procfs library is also used in other programs. I'd prefer to use the most natural data type (so probably integers for all?/most bcache fields) and do the type conversion to float64 in node_exporter.